src/udev/udev-node.c

```diff
@@ -465,8 +500,13 @@ int udev_node_remove(sd_device *dev) {
         assert(dev);
 
         /* remove/update symlinks, remove symlinks from name index */
-        FOREACH_DEVICE_DEVLINK(dev, devlink)
-                (void) link_update(dev, devlink, false);
+        FOREACH_DEVICE_DEVLINK(dev, devlink) {
+                r = link_update(dev, devlink, false);
+                if (r  == -ELOOP)
+                        log_device_debug_errno(dev, r, "Failed to update device symlinks: %m");
+                else if (r < 0)
+                        log_device_error_errno(dev, r, "Failed to update device symlinks: %m");
```

Comment:
the handling for the specific ELOOP and the generic case are exactly the same. I think the ELOOP case should say clearly what is going on:

```c
log_device_debug_errno(dev, r, "Failed to update device symlink correctly even after %u attempts, giving up.", LINK_UPDATE_MAX_RETRIES);
```

Reply:
oh, i see now that the lines are actually different: the log level is different. didn't see that first.

But I think we should log about ELOOP at least at LOG_WARN or so, too. It's a bit fishy after all if this happens. 

Hence maybe change the ELOOP case to log_device_warning_errno() and then use the log message string i suggested?

Reply:
> But I think we should log about ELOOP at least at LOG_WARN or so, too. It's a bit fishy after all if this happens.

Hmm, but the way this works right now it happens quite often. If there are two devices competing for the same symlink  with the same priority (e.g. in the first iteration of algorithm when non-existing db entries imply priority 0 for both of them) then we will always hit the loop limit. Basically they could be replacing each others symlink until one runs last and wins the race. Then we write db entries, one get higher priority and potential error gets correct in the second iteration.

Reply:
Maybe make it `LOG_NOTICE` and we'll downgrade it if turns out that we see the logs in real systems?

Reply:
I don't understand why need to go through this exercise. You would not see it if we updated udev db before running the algorithm but @poettering didn't want this. So we run algorithm twice and in first invocations it is very likely we will hit the loop limit.

Previously, we didn't log any failures from `link_update()` now all unexpected errors will be logged normally and looping will by logged at debug level. Again, this is not a anomaly but consequence of our design decision that udev db needs to be updated "last" even though we need that data in prior algorithms.

I tried to convince @poettering that this design, in my opinion, doesn't bring anything valuable because previously we could end up with wrong symlinks, now we potentially create wrong symlinks, update db and correct them. But from the user perspective, previously and after merge of this PR, return from `udev_device_is_initialized()` is still more or less meaningless and correctness is not guaranteed.